narwhal/types/src/primary.rs

```diff
@@ -281,6 +377,7 @@ impl Header {
     MallocSizeOf,
     Arbitrary,
 )]
+// TODO(arun): Should we create HeaderV1Digest?
```

Comment:
@mystenmark ^^

Reply:
I think HeaderDigest is fine.

Reply:
agree - the only thing here is that digest must be computed over the outer enum, so that the enum tag becomes part of the digest.